cpp/include/raft/core/mdarray.hpp

```diff
@@ -176,15 +317,24 @@ class mdarray {
   /**
    * @brief Get a mdspan that can be passed down to CUDA kernels.
    */
-  auto view() noexcept { return view_type(c_.data(), map_, cp_.make_accessor_policy()); }
+  view_type view() noexcept { return view_type(c_.data(), map_, cp_.make_accessor_policy()); }
```

Comment:
Sounds good, this looks clean.

Reply:
Hmm .. why do you need to make that check? If you want to emit error early, you can use:
``` c++
template <typename ArrayInterface, typename std::enable_if_t<is_mdspan_v<decltype(std::declval<A>().view())>* = nullptr>
auto my_func(ArrayInterface const& arr) {
}

Reply:
Having said that, I'm just trying to avoid c++ virtual table.  Might not be the most rational choice but from my experience template is easier to handle than virtual functions.

Reply:
Alternatively, `std::is_convertible` might help.

Reply:
`std:is_convertible` does not help since `mdarray` is a class-template and we do not know ahead of time if a template-type `T` could possibly a class-template of `mdarray`. I like your solution though, we can check if the `view()` returns a type which satisfies `is_mdspan_v`